### ISSUE TITLE
Wrong results for the grouping_operation.

### ISSUE THREAD

**Issue by HaiyongJiang** [erikwijmans/Pointnet2_PyTorch#17]
Dear @erikwijmans, thank you for your great work on the pytorch version of pointnet2. I tested the grouping_operation, but the result is not as expected. I checked the cuda implementation, and could not find any bug. Could you please check the code?

**Comment by erikwijmans** [comment]
I’m happy to take a look, but I need more information than “it’s wrong”. Do you have an input/output example pair?

**Comment by HaiyongJiang** [comment]
Hi @erikwijmans, thank you for your reply. I tested with the following snippets, and found the grouped points 'xyz1' did not match its corresponding points in 'xyz'. I hope you can reproduce it. 

    xyz_trans = torch.rand((1, 3, 32)).cuda()
    idx = torch.cat([torch.range(0,31).view(1, 32, 1),
                     torch.randint(0, 32, (1, 32, 5))], dim=-1).type(torch.long).cuda()
    grouped_xyz_src = grouping_operation( xyz_trans, idx)  # (B, 3, npoint, nsample)
    print("xyz1: ", grouped_xyz_src[:, :, :, 0])
    print("xyz_tran: ", xyz_trans)
    print("idxs: ", idx)

**Comment by erikwijmans** [comment]
idx should be an int tensor, not a long tensor.
